cumulus/xcm/xcm-emulator/src/lib.rs

```diff
@@ -615,12 +630,27 @@ macro_rules! decl_test_parachains {
 				// and have them ready for the next block
 				fn init() {
 					use $crate::{Chain, HeadData, Network, NetworkComponent, Hooks, Encode, Parachain, TestExt};
+					// Execute on_initialize hook
+					Self::on_initialze();
 					// Set the last block head for later use in the next block
 					Self::set_last_head();
 					// Initialize a new block
 					Self::new_block();
 					// Finalize the new block
 					Self::finalize_block();
+					// Execute on_finalize hook
+					Self::on_finalize();
+				}
+
+				fn on_initialize() {
+    				use $crate::{Chain, HeadData, Network, NetworkComponent, Hooks, Encode, Parachain, TestExt};
+
+    				Self::ext_wrapper(|| {
+        				let block_number = <Self as Chain>::System::block_number();
+						$(
+            				<<Self as Parachain>::ParachainSystem as Hooks<$crate::BlockNumber>>::$on_initialize_fn(block_number);
+        				)?
+    				});
```

Comment:
No need to create more methods, just to place `on_initialize_fn()` and `on_finalize_fn()` where we want them to be executed